src/neo/IO/Json/JPathToken.cs

```diff
@@ -209,10 +209,16 @@ private static void ProcessRecursiveDescent(ref JObject[] objects, Queue<JPathTo
             while (objects.Length > 0)
             {
                 if (identifier is null)
+                {
                     Descent(ref objects);
+                    results.AddRange(objects);
+                }
                 else
-                    Descent(ref objects, identifier);
-                results.AddRange(objects);
+                {
+                    var properties = objects.SelectMany(p => p.Properties);
+                    results.AddRange(properties.Where(p => p.Key == identifier).Select(p => p.Value));
+                    objects = properties.Select(p => p.Value).ToArray();
```

Comment:
This way we will miss some inner levels in arrays:
```
{ "s":
  { "name": 1,
    "sub":[{ "name": 2 }],
    "z": { "name": 3 }
}
```
`$.s..name` will be `[1,3]`, not `[2,3]`. I think this is because `Properties` return empty list for `JArray`.

Reply:
Fixed.